res/controllers/midi-components-0.0.js

```diff
@@ -311,45 +308,48 @@
         shift: function() {
             this.inKey = "hotcue_" + this.number + "_clear";
         },
-        getColor: function() {
-            if (this.colorIdKey !== undefined) {
-                return color.predefinedColorFromId(engine.getValue(this.group, this.colorIdKey));
-            } else {
-                return null;
-            }
-        },
         output: function(value) {
             var outval = this.outValueScale(value);
             // WARNING: outputColor only handles hotcueColors
             // and there is no hotcueColor for turning the LED
             // off. So the `send()` function is responsible for turning the
             // actual LED off.
-            if (this.colorIdKey !== undefined && outval !== this.off) {
-                this.outputColor(engine.getValue(this.group, this.colorIdKey));
+            if (this.colorKey !== undefined && outval !== this.off) {
+                this.outputColor(engine.getValue(this.group, this.colorKey));
             } else {
                 this.send(outval);
             }
         },
-        outputColor: function(id) {
-            var color = this.colors[id];
-            if (color instanceof Array) {
-                if (color.length !== 3) {
-                    print("ERROR: invalid color array for id: " + id);
-                    return;
-                }
-                if (this.sendRGB === undefined) {
-                    print("ERROR: no function defined for sending RGB colors");
-                    return;
-                }
-                this.sendRGB(color);
-            } else if (typeof color === "number") {
-                this.send(color);
+        outputColor: function(colorCode) {
+            // Sends the color from the colorCode to the controller. This
+            // method will not be called if no colorKey has been specified.
+            if (colorCode === undefined || colorCode < 0) {
```

Comment:
If not doing anything really preferable to just using the lower 3 bytes?

Reply:
I could also throw an error (which opens the script exception popup) if that's desired.

Reply:
It should match the behavior of the skin. In the c++ code we do nothing. I think we should not pop up a script error, because it is just an invalid value of the CO that can be come from everywhere.

Reply:
Please add ``` || colorCode > 0xFFFFFF ``` to protect the following code from insane values.